Add FreeBSD 10 support

danieldreier opened the pull request:
This updates params.pp with parameters that work with FreeBSD 10, lists FreeBSD 10 support in metadata.json, and adds zleslie/pkgng as a dependency. Without the pkgng module, puppet will try to use the gem provider to install a package named ruby, which obviously fails.

If adding a non-puppetlabs dependency is a problem, I can add a conditional statement in `init.pp` to skip the `ruby` package install if `osfamily == 'FreeBSD'`.

I'm not terribly experienced with FreeBSD, so I'm not sure if setting `$rubygems_package = 'ruby20-gems'` is reasonable. @xaque208 do you mind taking a look?

This is all in support of making [puppet-puppet](https://github.com/puppetlabs-operations/puppet-puppet) work on FreeBSD.

ghoneycutt commented on metadata.json:
```diff
@@ -69,6 +75,7 @@
   "dependencies": [
-    {"name":"puppetlabs/stdlib","version_requirement":">=3.2.0 <5.0.0"}
+    {"name":"puppetlabs/stdlib","version_requirement":">=3.2.0 <5.0.0"},
+    {"name":"zleslie/pkgng","version_requirement":">=0.2.5 <2.0.0"}
```
-1 to requiring zleslie/pkgng only because it is still < 1.0.0

Could you use the supported freebsd or ports providers?

xaque208 replied:
FWIW, the pkgng provider is set to be included in core. https://github.com/puppetlabs/puppet/pull/3276

hunner replied:
Yay, pkgng is in core. Could you remove this dependency, document it in the readme as a soft requirement for puppet < 4.1.0, and that it will work on 4.1.0 as-is?

xaque208 commented:
+1 at first glance.

danieldreier commented:
I've updated to remove the hard dependency on zleslie/pkgng and updated the readme to reflect it. If there's a better place to put it in the readme I'd be glad to move it. Also rebased.

igalic commented on metadata.json:
```diff
@@ -41,6 +41,12 @@
+      "operatingsystemrelease": [
+        "10"
+      ]
+    },
```
i don't think you should be updating metadata.json…

ghoneycutt replied:
This makes sense since @bmjen is adding support for a new platform.

nibalizer commented:
pkgng isn't being added as a dependency in metadata.json? Is this considered an implicit dependency?

bmjen commented:
@nibalizer It shouldn't be a hard dependency because pkgng is part of puppet 4.1+.

nibalizer commented:
Ok

igalic commented:
so, merge away?

bmjen commented:
@igalic do you still have qualms with the metadata?

ghoneycutt commented:
@nibalizer also not a hard dependency because you only need it on FreeBSD as opposed to it being an integral dependency such as stdlib
